lib/vapor/watch/supervisor.ex

```diff
@@ -0,0 +1,21 @@
+defmodule Vapor.Watch.Supervisor do
+  use DynamicSupervisor
+
+  def start_link(name: name) do
+    DynamicSupervisor.start_link(__MODULE__, :ok, name: name)
+  end
+
+  def sup_name(module) do
+    :"#{module}_watch_sup"
+  end
+
+  def start_child(module, state) do
```

Comment:
I think we have a case where if we exceed restart threshold for the supervisor, the watch supervisor will get restarted, but we'll not longer have any of the watches. So we'd have a runtime situation where Vapor is up and running, it's just not updating from any of its sources anymore. Maybe that's okay....